activerecord/lib/active_record/connection_adapters/abstract/connection_pool.rb

```diff
@@ -317,6 +317,9 @@ def register_pool(pool, frequency) # :nodoc:
           private
             def spawn_thread(frequency)
               Thread.new(frequency) do |t|
+                # Advise multi-threaded app servers to ignore this thread for
+                # the purposes of fork safety warnings
+                Thread.current[:fork_safe] = true
```

Comment:
It seems like this should be thread-local, rather than fiber-local.

```suggestion
                Thread.current.thread_variable_set(:fork_safe, true)
```